### DISCUSSION TITLE
Can I have conditional RLS on a table?  If so, how do I create this

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#13884] [Questions] [answered]
My application has 2 types of user - doctors and patients and every patient must have a doctor, and a doctor can have many patients.
The application has two tables - a "PROFILES" table for authenticated users with a role of either "doctor" or "patient"  and "CASE NOTES" table in which a "doctor" user creates records relating to his own "patients".  So, this is the easy part.  I can create RLS where a "doctor" can read/edit/delete records he has created and I do this by adding a field called for example "doctor_id" and a policy that user_id must equal "doctor_id' in order to gain access.
But my application also needs to enable each "patient" to read case notes relating to him/her and to add new records.  I think I therefore need an alternate RLS policy along the lines that IF an authenticated user has a role of PATIENT, he can read all records where field PATIENT_ID = authenticated user_id AND also can write a record where PATIENT_ID = authenticated user_id and patient's doctor_id = value in the table's doctor_id field. 
Am I on the right track?  And how do I create conditional RLS policies?  Maybe the ROLE itself can include a policy?

**Comment by GaryAustin1** [comment, marked as answer]
You can have multiple RLS policies.  So you can have one that allows a user to access their own role with userid=auth.uid() and then have another that does a join select on your profiles table to allow for doctors role to do more operations.
https://supabase.com/docs/guides/auth/row-level-security#policies-with-joins  Shows using joins.
https://github.com/orgs/supabase/discussions/6467 shows an is_admin function similar to what you might want.

**Reply by peppermintbunny** [reply]
Thank for this helpful answer, Gary
